qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/helper/TestData.java

```diff
@@ -684,4 +678,60 @@ public String toString() {
 		public String datetime;
 		public String uuid;
 	}
+
+	public static class PatientGenerator {
+
+		private static final String[] PATIENT_GIVEN_NAMES = {"Alexandre", "Achint",
+				"Brittany", "Burke", "Cesar", "Cosmin", "Daniel", "Darius",
+				"David", "Ellen", "Émerson", "Evan", "Fernando", "Gabou",
+				"Glauber", "Hamish", "Louise", "Mário", "Mark", "Natália", "Neil",
+				"Neissi", "Nelice", "Rafal", "Renee", "Wyclif"};
+		private static final String[] PATIENT_FAMILY_NAMES = {"Barbosa", "Sethi",
+				"Eddy", "Mamlin", "Vortmann", "Ioan", "Kayiwa", "Jazayeri",
+				"Walton", "Ball", "Hernandez", "Waters", "Freire", "Mandy",
+				"Ramos", "Fraser", "Sécordel", "Areias", "Goodrich", "Arsand",
+				"Craven", "Lima", "Heck", "Korytkowski", "Orser", "Luyima"};
+		private static final String[] PATIENT_BIRTH_MONTH = MONTHS;
+		private static final String[] PATIENT_BIRTH_DAY = DAYS;
+		private static final String[] PATIENT_BIRTH_YEAR = YEARS;
+		private static final String[] PATIENT_GENDER = {"Male", "Female"};
+
+		public static org.openmrs.contrib.qaframework.helper.TestPatient generateTestPatient() {
+			org.openmrs.contrib.qaframework.helper.TestPatient testPatient = new org.openmrs.contrib.qaframework.helper.TestPatient();
+			testPatient.givenName = randomArrayEntry(PATIENT_GIVEN_NAMES);
+			testPatient.middleName = "";
+			testPatient.familyName = randomArrayEntry(PATIENT_FAMILY_NAMES);
+			testPatient.gender = randomArrayEntry(PATIENT_GENDER);
+			testPatient.birthDay = randomArrayEntry(PATIENT_BIRTH_DAY);
+			testPatient.birthMonth = randomArrayEntry(PATIENT_BIRTH_MONTH);
+			testPatient.birthYear = randomArrayEntry(PATIENT_BIRTH_YEAR);
+			String suffix = randomSuffix();
+			testPatient.address1 = "Address1" + suffix;
+			testPatient.address2 = "Address2" + suffix;
+			testPatient.city = "City" + suffix;
+			testPatient.state = "State" + suffix; // TODO shorter string for State perhaps?
+			testPatient.country = "Country" + suffix; // TODO shorter string for Country
+			// perhaps?
```

Comment:
```suggestion
	testPatient.country = "Country" + suffix; // TODO shorter string for Country perhaps?
```

Reply:
@kdaud isnt  this duplication

Reply:
`testPatient.country = "Country" + suffix; // TODO shorter string for Country ` since its already on line 713...........

Reply:
@Muta-Jonathan cross check and do the needful !